desc/compute/_curve.py

```diff
@@ -639,6 +639,88 @@ def _x_sss_FourierXYZCurve(params, transforms, profiles, data, **kwargs):
     return data
 
 
+def _splinexyz_helper(f, p_knots, transforms, xq, kwargs, derivative):
+    """Used to compute XYZ coordinates for the SplineXYZCurve compute functions.
+
+    Parameters
+    ----------
+    f : list of ndarray
+        X, Y, Z function coords with shape (3, len(p_knots))
+    p_knots : ndarray
+        knots that come from params
+    transforms : dict
+        the transforms from the compute function
+    xq : ndarray
+        query points that come from s parameterization
+    kwargs : dict
+        the kwargs from the compute function
+    derivative : int
+        derivative order used for interpolation
+
+    Returns
+    -------
+    coords : ndarray
+        Interpolated XYZ coords with shape (3, len(xq))
+    """
+    method = kwargs.get("method", "cubic")
+    transforms["intervals"] = jnp.asarray(transforms["intervals"])
+    has_break_points = len(transforms["intervals"][0])
+    n_dim = 3
+
+    def inner_body(f, knots, period=None):
+        """Interpolation for spline curves."""
+        fq = [
+            interp1d(
+                xq,
+                knots,
+                f[i],
+                method=method,
+                derivative=derivative,
+                period=period,
+            )
+            for i in range(n_dim)
+        ]
+
+        return fq
+
+    def get_interval(arr, knots, istart, istop):
+        arr_temp = jnp.where(knots > knots[istop], arr[istop], arr)
+        arr_temp = jnp.where(knots < knots[istart], arr[istart], arr_temp)
+
+        return arr_temp
+
+    def body(i, fq):
+        istart, istop = transforms["intervals"][i]
+        # catch end-point
+        istop = jnp.where(istop == 0, -1, istop)
+
+        f_in_interval = [get_interval(f, full_knots, istart, istop) for f in full_f]
+        fq_temp = inner_body(f_in_interval, full_knots, period=None)
+
+        xq_range = (xq >= full_knots[istart]) & (xq <= full_knots[istop])
+        for i in range(n_dim):
+            fq = fq.at[i].set(jnp.where(xq_range, fq_temp[i], fq[i]))
+
+        return fq
+
+    if has_break_points:
+        # full_f, knots used in body()
+        # manually add endpoint for broken splines
+        full_knots = jnp.append(p_knots, p_knots[0] + 2 * jnp.pi)
+        full_f = [jnp.append(f[i], f[i][0]) for i in range(3)]
+        # query points for Xq, Yq, Zq
+        fq = jnp.zeros((n_dim, len(xq)))
+        fq = fori_loop(0, len(transforms["intervals"]), body, fq)
```

Comment:
I think this could be replaced by vmap + sum if you want. 
The idea is instead of modifying fq each time, you return fq_temp, but with the points outside the domain replaced with 0. So when vmapped you'd get back an array of shape(3, num_broken_segments, num_knots), where along the broken_segment dimension, there is only ever 1 nonzero element, so you can just do arr.sum(axis=1) to get the expected output.